core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -779,6 +779,22 @@ public void testAddRequiredColumnCaseInsensitive() {
         .hasMessage("Cannot add column, name already exists: ID");
   }
 
+  @Test
+  public void testAddMultipleRequiredColumnCaseInsensitive() {
+    Schema schema = new Schema(required(1, "id", Types.IntegerType.get()));
+
+    assertThatThrownBy(
+            () ->
+                new SchemaUpdate(schema, 1)
+                    .caseSensitive(false)
+                    .allowIncompatibleChanges()
```

Comment:
Minor: `allowIncompatibleChanges` isn't strictly required to repro this issue right? I think in this test case it was just set because we're adding required columns but if we were to remove this and just do `addOptionalColumn` for both "data" and "DATA", it should still just fail the same way.